src/vs/editor/contrib/stickyScroll/browser/stickyScrollProvider.ts

```diff
@@ -96,10 +96,44 @@ export class StickyLineCandidateProvider extends Disposable {
 		}
 	}
 
+	private customBinarySearch(model: readonly StickyOutlineElement[], startLine: number) {
+
+		// Binary search
+		let result;
+		let low = 0, high = model.length - 1;
+		while (low <= high) {
+			const mid = ((low + high) / 2) | 0;
+			const comp = model[mid].range?.startLineNumber as number - startLine;
+			if (comp < 0) {
+				low = mid + 1;
+			} else if (comp > 0) {
+				high = mid - 1;
+			} else {
+				result = mid;
+				break;
+			}
+		}
+		if (!result) { result = -(low + 1); }
+
+		// Update index
+		if (result === -1) {
+			result = 0;
+		} else if (result < 0) {
+			result = -result - 2;
+		}
+		return result;
+	}
+
 	public getCandidateStickyLinesIntersectingFromOutline(range: StickyRange, outlineModel: StickyOutlineElement, result: StickyLineCandidate[], depth: number, lastStartLineNumber: number): void {
+		if (!outlineModel || outlineModel.children.length === 0) {
```

Comment:
The `!outlineModel` condition should never be true or the argument typing is wrong, should be `outlineModel: StickyOutlineElement | undefined`